Sources/Interpreter/Memory.swift

```diff
@@ -282,6 +289,14 @@ public struct Memory {
     return .init(allocation: a, offset: 0)
   }
 
+  /// Allocates memory for type having type layout `t`.
```

Comment:
```suggestion
  /// Returns new memory for an instance of `t`.
```

It's usually better if you can avoid describing a thing in terms of words in the thing's name.